docs/creation.rst

```diff
@@ -0,0 +1,93 @@
+.. currentmodule:: xarray
+
+Creating and saving objects with units
+======================================
+
+Attaching units
+---------------
+.. ipython:: python
+    :suppress:
+
+    import xarray as xr
+
+Usually, when loading data from disk we get a :py:class:`Dataset` or
+:py:class:`DataArray` with units in attributes:
+
+.. ipython::
+
+    In [1]: ds = xr.tutorial.open_dataset("air_temperature")
+       ...: ds
+
+    In [2]: da = ds.air
+       ...: da
+
+In order to get :py:class:`pint.Quantity` instances, we can use the
+:py:meth:`Dataset.pint.quantify` or :py:meth:`DataArray.pint.quantify` methods:
+
+.. ipython::
+    :okexcept:
+
+    In [3]: ds.pint.quantify()
+
+As we can see, the dataset uses units like ``degree_north`` or ``degree_east``,
+which `pint`_ doesn't know about. To fix that, we can override the units of
+specific variables:
+
+.. ipython::
+
+    In [4]: ds.pint.quantify(lat="degree", lon="degree")
+
+    In [5]: da.pint.quantify({"lat": "degree", "lon": "degree"})
+
+Overriding works, even if there is no ``units`` attribute, so we could use this
+to attach units to a ordinary :py:class:`Dataset`:
+
+.. ipython::
+
+    In [6]: temporary_ds = xr.Dataset({"a": ("x", [0, 5, 10])}, coords={"x": [1, 2, 3]})
+       ...: temporary_ds.pint.quantify({"a": "m"})
+
+Of course, we could use :py:class:`pint.Unit` instances instead of strings to
+specify units, too. If we wanted to change the units of the data of a
+:py:class:`DataArray`, we could do so using the :py:attr:`DataArray.name`
+attribute:
+
+.. ipython::
+
+    In [7]: da.pint.quantify({da.name: "J", "lat": "degree", "lon": "degree"})
+
+However, `xarray`_ currently doesn't support `units in indexes`_, so the new units were set
+as attributes. To really observe the changes the ``quantify`` methods make, we
+have to first swap the dimensions:
+
+.. ipython::
+
+    In [8]: ds_with_units = ds.swap_dims({"lon": "x", "lat": "y"}).pint.quantify(
+       ...:     {"lat": "degree", "lon": "degree"}
+       ...: )
+       ...: ds_with_units
+
+    In [9]: da_with_units = da.swap_dims({"lon": "x", "lat": "y"}).pint.quantify(
+       ...:     {"lat": "degree", "lon": "degree"}
+       ...: )
+       ...: da_with_units
```

Comment:
While this totally makes sense as a workaround (replacing `degrees_north` and `degrees_east` with `degrees`), it unnecessarily breaks CF compliance and the ability to round trip the dataset, so it doesn't feel right to recommend it. I would rather recommend [adding `degrees_north` and `degrees_east` to the unit registry](https://github.com/Unidata/MetPy/blob/ee81f41baefdc9a3d41669a42a644c49f587121d/src/metpy/units.py#L53-L56), or at the very least adding a caveat here that this breaks the typical CF/NetCDF unit semantics (and at some point later link to another section on the CF compatible option).

Reply:
I agree, a caveat and a link to a CF compatible option would be good. However, I'm reluctant to recommend defining those as an alias of `degree` since the suffix makes them physically different (and `pint` wouldn't complain when converting `degrees_east` to `degrees_north`).

Ideally, a library like `cf-xarray` would do something with the direction and then set the units to `degree`. That way, we wouldn't have to deal with these units.

Reply:
after thinking about this some more, I decided to avoid the issue altogether by defining a synthetic dataset. I still think that the units should be preprocessed by something that knows about the CF conventions, but that's something we can discuss in a different issue.

Reply:
> after thinking about this some more, I decided to avoid the issue altogether by defining a synthetic dataset. I still think that the units should be preprocessed by something that knows about the CF conventions, but that's something we can discuss in a different issue.

This is a bit of a thorny issue depending on your perspectives on unit correctness, so I definitely agree that discussion should happen elsewhere. (see https://github.com/xarray-contrib/pint-xarray/issues/26)